File: app/src/main/java/com/nlapin/youthsongs/BaseContract.java
package com.nlapin.youthsongs;

import java.util.Collection;

public interface BaseContract {

    interface BaseView {

        void showProgressBar();

        void hideProgressBar();
    }

    interface BasePresenter<V extends BaseView> {

        void attachView(V v);

        void detachView();

        void showError();

        V getView();

        boolean isViewAttached();
    }
}

File: app/src/main/java/com/nlapin/youthsongs/domain/BaseContract.kt
package com.nlapin.youthsongs.domain

interface BaseContract {
    interface BaseView {
        fun onError()
    }

    interface BasePresenter<V : BaseView> {
        fun attachView(view: V)
        fun detachView()
        fun getView(): V?
        fun isViewAttached(): Boolean
    }
}
